### PR TITLE
feat: zopa footer accept cobranding

### PR DESCRIPTION
**Pull request by zopa-charlotte-walker**
Cobranding in footer

### CODE REVIEW

File: src/components/molecules/ZopaFooter/__snapshots__/ZopaFooter.test.tsx.snap [hunk 1]
```diff
@@ -219,6 +219,7 @@ exports[`<ZopaFooter /> renders the component with no a11y violations 1`] = `
   font-weight: 400;
   -webkit-text-decoration: none;
   text-decoration: none;
+  color: #3B46C4;
```
Comment (zopa-charlotte-walker) [inline src/components/molecules/ZopaFooter/__snapshots__/ZopaFooter.test.tsx.snap:222]: no visual change

File: src/components/molecules/ZopaFooter/ZopaFooter.tsx [hunk 2]
```diff
@@ -402,9 +404,19 @@ const ZopaFooter = ({
...
-  const theme = useThemeContext();
+  const themeContext = useThemeContext();
+  let theme: AppTheme;
+  if (isCobranded) {
+    theme = zopaTheme;
+  } else {
+    theme = themeContext;
+  }
+
+  console.log('mainCobrandedLegalCopy', mainCobrandedLegalCopy);
```
Comment (MichelleXBai) [inline src/components/molecules/ZopaFooter/ZopaFooter.tsx:419]: Remove

File: src/components/molecules/ZopaFooter/ZopaFooter.tsx [hunk 3]
```diff
@@ -421,18 +433,26 @@ const ZopaFooter = ({
...
+                ) : (
+                  <MainZopaLegalCopy />
+                )}
+                {additionalCopy.map((copy, i) => (
+                  <Text
+                    as="p"
+                    color={theme.footer.legalBlock.color}
+                    size="small"
+                    key={i}
+                    className={i > additionalCopy.length ? 'mb-0' : 'mb-4'}
```
Comment (MichelleXBai) [inline src/components/molecules/ZopaFooter/ZopaFooter.tsx:450]: What are you trying to map here?

Comment (MichelleXBai) [reply src/components/molecules/ZopaFooter/ZopaFooter.tsx:450]: `i` can't be greater than `additionalCopy.length - 1` ?

**Pull request merged**